appengine/flexible/TwelveFactor/Startup.cs

```diff
@@ -0,0 +1,172 @@
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var projectId = GetProjectId();
+            // Add framework services.Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService
+            services.AddMvc();
+            // Enables Stackdriver Trace.
+            services.AddGoogleTrace(projectId);
+            // Sends Exceptions to Stackdriver Error Reporting.
+            services.AddGoogleExceptionLogging(projectId, GetServiceName(), GetVersion());
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            var projectId = GetProjectId();
+
+            // Only use Console and Debug logging during development.
+            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+            loggerFactory.AddDebug();
+            // Send logs to Stackdriver Logging.
+            loggerFactory.AddGoogle(projectId);
+            
+            _cloudStorage.Logger = loggerFactory.CreateLogger<CloudStorageFileProvider>();
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+                app.UseBrowserLink();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Home/Error");
+                app.UseGoogleExceptionLogging();
+            }
+
+            app.UseStaticFiles();
+            app.UseGoogleTrace();
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+
+        private string GetProjectId()
+        {
+            var instance = Google.Api.Gax.Platform.Instance();
+            var projectId = instance?.ProjectId ?? Configuration["Google:ProjectId"];
+            if (string.IsNullOrEmpty(projectId))
+            {
+                throw new Exception(
+                    "The logging, tracing and error reporting libraries need a project ID. " +
+                    "Update appsettings.json by setting the ProjectId property with your " +
+                    "Google Cloud Project ID, then recompile.");
+            }
+            return projectId;
+        }
+
+        private string GetServiceName()
+        {
+            var instance = Google.Api.Gax.Platform.Instance();
+            // An identifier of the service. See https://cloud.google.com/error-reporting/docs/formatting-error-messages#FIELDS.service.
+            var serviceName =
+                instance?.GaeDetails?.ServiceId ??
+                Configuration["Google:ErrorReporting:ServiceName"];
+            if (string.IsNullOrEmpty(serviceName))

```

Comment:
This is quite harsh, I think that we need to rethink our stance and perhaps log a warning?